### ISSUE TITLE
Ability to export `pixel_unshuffle` to ONNX

### ISSUE THREAD

**Issue by jkyl** [pytorch/pytorch#51181]
## 🚀 Feature
Request support for exporting the new `nn.PixelUnshuffle` / `F.pixel_unshuffle` operation to ONNX using `torch.onnx.export`.

## Motivation
The "PixelUnshuffle" operation, also known as "SpaceToDepth", was recently added to torch with [this feature request](https://github.com/pytorch/pytorch/issues/2456). This operation is the inverse of the "PixelShuffle" / "DepthToSpace" operation, and both operations have been [supported by ONNX](https://github.com/onnx/onnx/blob/master/docs/Operators.md) since at least opset 9. However, torch only supports exporting PixelShuffle, and *not* Pixel**Un**shuffle, to ONNX. This precludes users' ability to deploy models trained in torch that include PixelUnshuffle layers to ONNX (and other) runtimes. Such models include convolutional U-nets, autoencoders, super-resolution models, and many others that require efficient lossless spatial re-sampling. 

Example:
```python
import torch

layer = torch.nn.PixelUnshuffle(2)

x = torch.rand(1, 1, 2, 2)

assert (layer(x) == x.view(1, 4, 1, 1)).all().item()  # passes

for opset_version in range(9, 14):
    try:
        torch.onnx.export(layer, x, "./pixel_unshuffle.onnx", opset_version=opset_version)
    except RuntimeError as err:
        print(err)
```
Prints for every version:
```
Exporting the operator pixel_unshuffle to ONNX opset version <version> is not supported. Please open a bug to request ONNX export support for the missing operator.
```

## Pitch
In order to achieve parity with PixelShuffle, and in order to support deploying the many vision models that require it, I request that torch developers please add support for exporting the PixelUnshuffle operation to ONNX.

## Alternatives
This feature is in fact present in tensorflow and tf2onnx, so I am currently using that framework for my particular use case. This induces hassle and stress among my group, as we would prefer to stick to a single framework.

## Additional context
I was advised to file a feature request by @mruberry [here](https://github.com/pytorch/pytorch/issues/2456#issuecomment-759835306).

cc @houseroad @spandantiwari @lara-hdr @BowenBao @neginraoof @gmagogsfm

**Comment by jkyl** [comment]
Thanks all for the response. Is there any chance you could provide an ETA, or just a general sense of the priority for this feature?

**Comment by jkyl** [comment]
Am I to understand that this feature is low-priority and that no work has yet been planned?

**Comment by spandantiwari** [comment]
@jkyl - We have put this op on our list to track, but we don't have an ETA for this. Please consider submitting a PR adding support for this op.

**Comment by jkyl** [comment]
@spandantiwari thank you for the update!

**Comment by jkyl** [comment]
Here's a small patch that worked for me: 
```diff
diff --git a/torch/onnx/symbolic_opset11.py b/torch/onnx/symbolic_opset11.py
index 6f18253a9e..c301888e7f 100644
--- a/torch/onnx/symbolic_opset11.py
+++ b/torch/onnx/symbolic_opset11.py
@@ -207,6 +207,14 @@ def pixel_shuffle(g, self, upscale_factor):
     return g.op("DepthToSpace", self, blocksize_i=upscale_factor, mode_s="CRD")
 
 
+@parse_args('v', 'i')
+def pixel_unshuffle(g, self, downscale_factor):
+    rank = sym_help._get_tensor_rank(self)
+    if rank is not None and rank != 4:
+        return _unimplemented("pixel_unshuffle", "only support 4d input")
+    return g.op("SpaceToDepth", self, blocksize_i=downscale_factor)
+
+
 def _interpolate(name, dim, interpolate_mode):
     return sym_help._interpolate_helper(name, dim, interpolate_mode)
```

I will submit a PR with this and a test. Any guidance would be much appreciated.

**Comment by jkyl** [comment]
The test that I wrote is failing in CI so I will need to put some more work into this before it's ready. Will update when finished.

**Comment by Tsubajashi** [comment]
Any news on this?

**Comment by haydn-jones** [comment]
@Tsubajashi  Left an update in the PR associated with this ( #53539 ) with some of my findings.

**Comment by Goose-Bomb** [comment]
It seems that the channel order of `torch.nn.PixelUnShuffle` and onnx's `SpaceToDepth` is different. Too bad that both of them don't have options to control data layout like that of `torch.nn.PixelShuffle` or onnx's `DepthToSpace`.

**Comment by shubhambhokare1** [comment]
PR https://github.com/pytorch/pytorch/pull/72449 should add pixel_unshuffle support. We will have to wait for ONNX's mode_s implementation in SpaceToDepth operator until we can add an updated implementation.

**Comment by BowenBao** [comment]
Hi @jkyl @haydn-jones @Tsubajashi ,

We’ve gone ahead and closed this issue because it has been resolved via https://github.com/pytorch/pytorch/pull/72449
If you still believe this issue is relevant, please feel free to reopen the issue and we will triage it as necessary. Please specify in a comment any updated information you may have so that we can address it effectively. We encourage you to try the latest pytorch-preview (nightly) version to see if it has resolved the issue.

Thanks,
ONNX Converter team
